pandas/core/arrays/datetimelike.py

```diff
@@ -667,12 +667,10 @@ def _validate_listlike(self, value, allow_object: bool = False):
                     msg = self._validation_error_message(value, True)
                     raise TypeError(msg) from err
 
-        # Do type inference if necessary up front (after unpacking
-        # NumpyExtensionArray)
+        # Do type inference if necessary up front
         # e.g. we passed PeriodIndex.values and got an ndarray of Periods
-        value = extract_array(value, extract_numpy=True)
-        value = pd_array(value)
-        value = extract_array(value, extract_numpy=True)
+        value = sanitize_array(value, index=None)
+        value = ensure_wrapped_if_datetimelike(value)
```

Comment:
This is essentially the default input sanitation/inference we do for the Series constructor (and other places), so switched to use that here instead of `pd.array(..)` (which doesn't use the default dtypes for all types)